community/kernel/src/main/java/org/neo4j/unsafe/impl/batchimport/store/BatchingNeoStores.java

```diff
@@ -73,90 +77,147 @@
-                try
-                {
-                    pageCache.close();
-                }
-                catch ( Exception e )
-                {
-                    // Oddly enough we can't close the page cache, how to communicate this? Here we add as suppressed
-                    ise.addSuppressed( e );
-                }
-            }
-            throw ise;
+            // OK so the db probably exists
         }
-        try
+        catch ( IOException e )
         {
-            neoStores.rebuildCountStoreIfNeeded();
+            // It's OK
+            return false;
         }
-        catch ( IOException e )
+
+        try ( NeoStores stores = newStoreFactory( DEFAULT_NAME ).openNeoStores( StoreType.NODE, StoreType.RELATIONSHIP ) )
+        {
+            return stores.getNodeStore().getHighId() > 0 || stores.getRelationshipStore().getHighId() > 0;
+        }
+    }
+
+    /**
+     * Called when expecting a clean {@code storeDir} folder and where a new store will be created.
+     * This happens on an initial attempt to import.
+     *
+     * @throws IOException on I/O error.
+     * @throws IllegalStateException if {@code storeDir} already contains a database.
+     */
+    public void createNew() throws IOException
+    {
+        if ( databaseExistsAndContainsData( pageCache, storeDir ) )
         {
-            throw new UnderlyingStorageException( e );
+            throw new IllegalStateException( storeDir + " already contains data, cannot do import here" );
         }
+
+        instantiateStores();
         neoStores.getMetaDataStore().setLastCommittedAndClosedTransactionId(
                 initialIds.lastCommittedTransactionId(), initialIds.lastCommittedTransactionChecksum(),
                 BASE_TX_COMMIT_TIMESTAMP, initialIds.lastCommittedTransactionLogByteOffset(),
                 initialIds.lastCommittedTransactionLogVersion() );
-        this.propertyKeyRepository = new BatchingPropertyKeyTokenRepository(
-                neoStores.getPropertyKeyTokenStore() );
-        this.labelRepository = new BatchingLabelTokenRepository(
-                neoStores.getLabelTokenStore() );
-        this.relationshipTypeRepository = new BatchingRelationshipTypeTokenRepository(
-                neoStores.getRelationshipTypeTokenStore() );
+    }
+
+    /**
+     * Called when expecting a previous attempt/state of a database to open, where some store files should be kept,
+     * but others deleted. All temporary stores will be deleted in this call.
+     *
+     * @param mainStoresToKeep {@link Predicate} controlling which files to keep, i.e. {@code true} means keep, {@code false} means delete.
+     * @param tempStoresToKeep {@link Predicate} controlling which files to keep, i.e. {@code true} means keep, {@code false} means delete.
+     * @throws IOException on I/O error.
+     */
+    public void pruneAndOpenExistingStore( Predicate<StoreType> mainStoresToKeep, Predicate<StoreType> tempStoresToKeep ) throws IOException
```

Comment:
this looks like a method that is quite dangerous to keep; it brings too much tempt and calling it can have really harsh consequences

Reply:
It will be used in an upcoming feature. I know it doesn't need to be here right now... but I don't see the need to go through the work of moving it to another PR later.

Also, import and BatchingNeoStores as a whole is already dangerous lands to be in.